scripts/image_app.py

```diff
@@ -0,0 +1,45 @@
+import streamlit as st
+from openai import OpenAI
+from prediction_market_agent_tooling.markets.markets import (
+    MarketType,
+    get_binary_markets,
+)
+
+st.title("Prediction market thumbnail generator")
+
+markets = get_binary_markets(42, MarketType.OMEN)
+
+custom_question_input = st.checkbox("Provide a custom question", value=False)
+question = (
+    st.text_input("Question")
+    if custom_question_input
+    else st.selectbox("Select a question", [m.question for m in markets])
+)
+
+prompt_template = st.text_area(
+    "Prompt",
+    value="""Create a thumbnail image for the following prediction market '{market}'.
+Don't write any text on the image.""",
+)
+
+market = (
+    [m for m in markets if m.question == question][0]
+    if not custom_question_input
+    # If custom question is provided, just take some random market and update its question.
+    else markets[0].model_copy(update={"question": question, "current_p_yes": 0.5})
+)
+
+with st.spinner("Generating image..."):
+    response = OpenAI().images.generate(
```

Comment:
Also, can we use `dall-e-2` since it's just a preview and thus save some $?

Reply:
Nope, dall-e-2 doesn't really work here, for example:

![image](https://github.com/gnosis/prediction-market-agent/assets/15619339/ee546f58-872d-4847-9252-2258245acd59)

vs. the same market in dall-e-2:

![image](https://github.com/gnosis/prediction-market-agent/assets/15619339/672d86f7-ab0f-464b-b2e7-af05141903da)

Reply:
That's impressive. Closing my own comment :laughing: